docs/developer-setup.rst

```diff
@@ -0,0 +1,208 @@
+===============
```

Comment:
Looks good. 

One ask: Please add a reference to this file in index.rst.

Reply:
Done!